Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/ComponentModeActionHandler.cpp

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzToolsFramework/ComponentMode/ComponentModeActionHandler.h>
+
+#include <AzToolsFramework/ActionManager/Action/ActionManagerInterface.h>
+#include <AzToolsFramework/ActionManager/Menu/MenuManagerInterface.h>
+#include <AzToolsFramework/ActionManager/HotKey/HotKeyManagerInterface.h>
+#include <AzToolsFramework/ComponentMode/EditorComponentModeBus.h>
+#include <AzToolsFramework/Editor/ActionManagerUtils.h>
+
+static constexpr AZStd::string_view EditorMainWindowActionContextIdentifier = "o3de.context.editor.mainwindow";
+
+static constexpr AZStd::string_view ComponentModeChangedUpdaterIdentifier = "o3de.updater.onComponentModeChanged";
+
+static constexpr AZStd::string_view EditMenuIdentifier = "o3de.menu.editor.edit";
+
+namespace AzToolsFramework
+{
+    ComponentModeActionHandler::ComponentModeActionHandler()
+    {
+        if (IsNewActionManagerEnabled())
+        {
+            AzFramework::EntityContextId editorEntityContextId = AzFramework::EntityContextId::CreateNull();
+            EditorEntityContextRequestBus::BroadcastResult(editorEntityContextId, &EditorEntityContextRequests::GetEditorEntityContextId);
+
+            ActionManagerRegistrationNotificationBus::Handler::BusConnect();
+            ComponentModeFramework::EditorComponentModeNotificationBus::Handler::BusConnect(editorEntityContextId);
```

Comment:
update: This should instead likely be `ViewportEditorModeNotifications` and not specifically `EditorComponentModeNotificationBus`

Reply:
For this I'd listen to `ActiveComponentModeChanged` on the `EditorComponentNotificationBus`, you might need that and the other I mentioned but you shouldn't need to add a new notification (just listen for both of those)

Reply:
Forgive me, I'm a little confused now :D I have searched for `EditorComponentNotificationBus` in the whole solution and I get no hits. Do you mean `EditorComponentModeNotificationBus`? Cause if that's the case, current implementation is using that already (and I have reverted the addition of the new notifications)